scm/github.go

```diff
@@ -661,6 +661,88 @@ func (s *GithubSCM) UpdateIssueComment(ctx context.Context, opt *IssueCommentOpt
 	return nil
 }
 
+// CreateCourse creates repositories and teams for a new course.
+func (s *GithubSCM) CreateCourse(ctx context.Context, opt *NewCourseOptions) ([]*Repository, error) {
+	org, err := s.GetOrganization(ctx, &GetOrgOptions{ID: opt.OrganizationID})
+	if err != nil {
+		return nil, err
+	}
+	repos, err := s.GetRepositories(ctx, org)
+	if err != nil {
+		return nil, err
+	}
+	if IsDirty(repos) {
```

Comment:
If we need IsDirty elsewhere, it should obviously be exported, but if not it can be made unexported.

Also, we should move the call to get repositories into isDirty and only pass in the organization.

Reply:
And ctx

Reply:
IsDirty is still used by the API `GetOrganization` method (which is being called when checking the organization before giving a user access to the front-end form to create a new course). I think this part can also be eventually moved to the `scm` package, but probably in another PR. This is why `IsDirty` has to stay exported for now.

I can see two ways to change `IsDirty` in the proposed way so that it still can be used by the API method (for now) :
1. To add it to the general scm methods ( `(scm.SCM).IsDirty(ctx, org)` )
2. Or to pass an scm client to it (`IsDirty(context.Context, scm.SCM, *qf.Organization)`)

Which solution is best or are there any better options?